include/inexor/vulkan-renderer/wrapper/resource_descriptor.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <vulkan/vulkan_core.h>
+
+#include <string>
+#include <vector>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+/// @brief A class for descriptor management.
+/// Shader access of data is managed through descriptors.
+/// Descriptors are organized in descriptor sets.
+/// Descriptors sets are described through their descriptor set layout.
+/// Descriptor sets are allocated from descriptor pools.
+class ResourceDescriptor {
+private:
+    std::string m_name;
+    VkDevice m_device{VK_NULL_HANDLE};
+    VkDescriptorPool m_descriptor_pool{VK_NULL_HANDLE};
+    VkDescriptorSetLayout m_descriptor_set_layout{VK_NULL_HANDLE};
+    std::vector<VkDescriptorSetLayoutBinding> m_descriptor_set_layout_bindings{};
+    std::vector<VkWriteDescriptorSet> m_write_descriptor_sets{};
+    std::vector<VkDescriptorSet> m_descriptor_sets{};
+    std::uint32_t m_swapchain_image_count = 0;
+
+public:
+    // @note Creates a descriptor pool.
+    ResourceDescriptor(VkDevice device, const std::uint32_t swapchain_image_count,
+                       const std::initializer_list<VkDescriptorType> pool_types,
+                       std::vector<VkDescriptorSetLayoutBinding> &layout_bindings,
+                       std::vector<VkWriteDescriptorSet> &descriptor_writes, const std::string &name);
```

Comment:
```suggestion
    ResourceDescriptor(VkDevice device, std::uint32_t swapchain_image_count,
                       std::initializer_list<VkDescriptorType> pool_types,
                       const std::vector<VkDescriptorSetLayoutBinding> &layout_bindings,
                       const std::vector<VkWriteDescriptorSet> &descriptor_writes, const std::string &name);
```
The vectors should be passed by const reference, and `const` on value parameters in the header does nothing.

Reply:
Done.